File: src/main/java/com/hedvig/backoffice/services/tickets/dto/TicketDto.java
package com.hedvig.backoffice.services.tickets.dto;

import lombok.Value;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.UUID;

@Value
public class TicketDto {
  private UUID id;
  private String assignedTo;
  private Instant creationDate;
  private String memberId;
  private String referenceId;
  private String createdBy;
  @Min(0)
  @Max(1)
  private float priority;
  private TicketType type;
  private LocalDate remindNotificationDate;
  private LocalTime remindNotificationTime;
  private String remindMessage;
  private String description;
  private TicketStatus status;

}

File: src/main/java/com/hedvig/backoffice/services/tickets/dto/TicketDto.kt
package com.hedvig.backoffice.services.tickets.dto

import java.time.Instant
import java.time.LocalDate
import java.time.LocalTime
import java.util.*
import javax.validation.constraints.Max
import javax.validation.constraints.Min

data class TicketDto(
    val id: UUID,
    val assignedTo: String? = null,
    val createdAt: Instant,
    val memberId: String? = null,
    val referenceId: String? = null,
    val createdBy: String,
    @Min(0)
    @Max(1)
    val priority: Float = 0f,
    val type: TicketType,
    val remindNotificationDate: LocalDate? = null,
    val remindNotificationTime: LocalTime? = null,
    val remindMessage: String? = null,
    val description: String? = null,
    val status: TicketStatus
)
